core/src/main/java/org/projectnessie/cel/interpreter/Activation.java

```diff
@@ -114,6 +118,32 @@ public String toString() {
     }
   }
 
+  /** functionActivation which implements Activation and a provider of named values. */
+  final class FunctionActivation implements Activation {
+    private final Function<String, Object> provider;
+
+    FunctionActivation(Function<String, Object> provider) {
+      this.provider = provider;
+    }
+
+    /** Parent implements the Activation interface method. */
+    @Override
+    public Activation parent() {
+      return null;
+    }
+
+    /** ResolveName implements the Activation interface method. */
+    @Override
+    public Object resolveName(String name) {
+      return provider.apply(name);
```

Comment:
`MapActivation` has some logic for memoizing lazy values provided by a `Supplier`. `FunctionActivation` lazily computes them by default, but should there be similar logic for memoizing some or all calls to the provider?